File: app/enums.py
from enum import Enum

class UserRole(str, Enum):
    ADMIN = "admin"
    ORGANIZER = "organizer"
    ANALYST = "analyst"
    REPORTER = "reporter"
    PERSONAL = "personal"
    STUDENT = "student"
    MODERATOR = "moderator"

class MeetingStatus(str, Enum):
    SCHEDULED = "scheduled"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

File: echoflow_ui/src/types/enums.ts
export const UserRole = {
  ADMIN: "admin",
  ORGANIZER: "organizer",
  ANALYST: "analyst",
  REPORTER: "reporter",
  PERSONAL: "personal",
  STUDENT: "student",
  MODERATOR: "moderator",
} as const;

export type UserRole = (typeof UserRole)[keyof typeof UserRole];

// Optional utility
export function isUserRole(value: any): value is UserRole {
  return Object.values(UserRole).includes(value);
}